ibm/service/cos/resource_ibm_cos_backup_vault_test.go

```diff
@@ -0,0 +1,337 @@
+
+	return fmt.Sprintf(`
+	resource "ibm_cos_backup_vault" "backup-vault" {
+		backup_vault_name           = "%s"
+		service_instance_id  = "%s"
+		region  = "%s"
+		activity_tracking_management_events = "%t"
+	}
+		
+	`, bucketVaultName, instance_id, region, managementEvents)
+}
+
+func testAccCheckIBMCosBackup_Vault_Create_With_Metrics_Monitoring_Valid(instance_id string, bucketVaultName string, region string, usageMetrics bool) string {
+
+	return fmt.Sprintf(`
+	resource "ibm_cos_backup_vault" "backup-vault" {
+		backup_vault_name           = "%s"
+		service_instance_id  = "%s"
+		region  = "%s"
+		metrics_monitoring_usage_metrics = "%t"
+	}
+		
+	`, bucketVaultName, instance_id, region, usageMetrics)
+}
+
+func testAccCheckIBMCosBackup_Vault_Create_With_KP_Valid(instance_id string, bucketVaultName string, region string, kmsKeyCrn string) string {
+
+	return fmt.Sprintf(`
+	resource "ibm_cos_backup_vault" "backup-vault" {
+		backup_vault_name           = "%s"
+		service_instance_id  = "%s"
+		region  = "%s"
+		kms_key_crn = "%s"
+	}
+		
+	`, bucketVaultName, instance_id, region, kmsKeyCrn)
+}
+
+func testAccCheckIBMCosBackup_Vault_Create_With_AT_Enabled_MM_Disabled_Valid(instance_id string, bucketVaultName string, region string, managementEvents bool, usageMetrics bool) string {
+
+	return fmt.Sprintf(`
+	resource "ibm_cos_backup_vault" "backup-vault" {
+		backup_vault_name           = "%s"
+		service_instance_id  = "%s"
+		region  = "%s"
+		activity_tracking_management_events = "%t"
+		metrics_monitoring_usage_metrics = "%t"
+	}
+		
+	`, bucketVaultName, instance_id, region, managementEvents, usageMetrics)
+}
+
+func testAccCheckIBMCosBackup_Vault_Create_With_AT_Disabled_MM_Enabled_Valid(instance_id string, bucketVaultName string, region string, managementEvents bool, usageMetrics bool) string {
+
+	return fmt.Sprintf(`
+	resource "ibm_cos_backup_vault" "backup-vault" {
+		backup_vault_name           = "%s"
+		service_instance_id  = "%s"
+		region  = "%s"
+		activity_tracking_management_events = "%t"
+		metrics_monitoring_usage_metrics = "%t"
+	}
+		
+	`, bucketVaultName, instance_id, region, managementEvents, usageMetrics)
+}
+
+func testAccCheckIBMCosBackup_Vault_Create_Multiple_Valid(instance_id string, bucketVaultName string, bucketVaultName2 string, region string) string {
+
+	return fmt.Sprintf(`
+	resource "ibm_cos_backup_vault" "backup-vault" {
+		backup_vault_name           = "%s"
+		service_instance_id  = "%s"
+		region  = "%s"
+	}
+
+	resource "ibm_cos_backup_vault" "backup-vault2" {
+		backup_vault_name           = "%s"
+		service_instance_id  = "%s"
+		region  = "%s"
+	}
+		
+	`, bucketVaultName, instance_id, region, bucketVaultName2, instance_id, region)
+}
+
+func testAccCheckIBMCosBackup_Vault_Create_Multiple_Invalid(instance_id string, bucketVaultName string, region string) string {
```

Comment:
what do you mean by multiple invalid, it seems like you are testing creating the backup vaults with the same name, in that case change the test case name to reflect that

Reply:
Done